### ISSUE TITLE
Labels are gone after applying dtw_dismat

### ISSUE THREAD

**Issue by jaySf** [maxar/IncDTW#1]
Hi Maximilian,
it seems we're encountering an issue with the  `dtw_dismat()` function. When we calculate a matrix the labels will be replaced by numbers.

A small reproducible example would be:

```
dtw_dismat(L, dist_method="norm2", return_matrix=FALSE)
```
As one can see the character labels A-D are gone.

Data:
```
L <- list(A = structure(1:50, .Dim = c(50L, 1L)), B = structure(1:75, .Dim = c(75L, 
1L)), C = structure(25:49, .Dim = c(25L, 1L)), D = structure(1:50, .Dim = c(50L, 
1L)))
```
The labels are used for plotting a dendrogram, otherwise there will be a conflict with "dendro" and "dendrogram" objects. See also these consecutive questions on Stack Overflow:

- [https://stackoverflow.com/q/54616539/6574038](https://stackoverflow.com/q/54593085/6574038)
- [https://stackoverflow.com/q/54616539/6574038](https://stackoverflow.com/q/54616539/6574038)

It would be great if you could have a look into this.
Cheers

**Comment by maxar** [comment]
Thanks for pointing me to this new feature. I will consider it in the next release of IncDTW. In the meanwhile a quick fix would be to do the following:

```
a <- matrix(1:9, 3, 3)
a <- a + t(a)
as.dist(a)
b <- usedist::dist_setNames(a, letters[1:3])
b

# where dist_setnames() does the following:
dm <- as.matrix(d)
dimnames(dm) <- list(nm, nm)
[stats::as.dist(dm)]
```
So dist_setnames() converts the dist.object to a matrix, which is fine for small lists of time series, but maybe a problem if the distance matrix gets big.

I also answered here:
[https://stackoverflow.com/questions/54593085/ggdendrogram-labels/54626400#54626400 ](https://stackoverflow.com/questions/54593085/ggdendrogram-labels/54626400#54626400 )

**Comment by jaySf** [comment]
Thanks for your prompt reply, this is a nice workaround for our application that works perfectly!
Cheers

**Comment by maxar** [comment]
Fixed in version 1.1.2 on CRAN
